runtime/lua/vim/lsp/buf.lua

```diff
@@ -47,6 +47,17 @@ local function request_with_opts(name, params, opts)
   request(name, params, req_handler)
 end
 
+--- Calls client.request with the given `opts`.
+local function client_request_with_options(client, name, params, opts, bufnr)
```

Comment:
is this different from `request_with_opts` which was merged in https://github.com/neovim/neovim/pull/28637 ?